### PR TITLE
hng64.cpp: Hyper NeoGeo 64 video updates

### PR DESCRIPTION
**Pull request by DavidHaywood**
- filter out most (but not all) the bad polygons in roadedge / xrally by handling display list more correctly
- fixed 4bpp texture handling (they were being treated as 'low resolution' textures, throwing away 3/4 of the data) (used extensively for background details on sams64/sams64_2)
- added 4bpp texture decode for easy viewing of the 4bpp texture pages, handy for debugging
- fixed some texture palette issues connected to the 4bpp textures and incorrect enable bits being used (wheels+windscreen palette in racing games for example)
- found a flag which seems to enable/disable the backface culling, improves bbust2 school bus without breaking roadedge
- a few small tidyups / cleanups (disabled some printf/logerrors that would be heavily spammed at times)
- add ADK as the developer of bbust2 (SNK/ADK is credited in the ending, even if not on the title screen)

### CODE REVIEW

File: src/mame/snk/hng64.cpp [hunk 1]
```diff
@@ -1720,20 +1720,48 @@ static const gfx_layout hng64_16x16x8_spritelayout =
 };
 
 static const uint32_t texlayout_xoffset[1024] = { STEP1024(0,8) };
-static const uint32_t texlayout_yoffset[512] = { STEP512(0,8192) };
-static const gfx_layout hng64_texlayout =
+static const uint32_t texlayout_yoffset[1024] = { STEP1024(0,8192) };
+
+static const uint32_t texlayout_xoffset_4[1024] = {
```
Comment (happppp) [inline src/mame/snk/hng64.cpp:1725]: Can this be precalculated?

File: src/mame/snk/hng64_v.cpp [hunk 2]
```diff
@@ -569,7 +569,7 @@ void hng64_state::hng64_drawtilemap(screen_device &screen, bitmap_rgb32 &bitmap,
 		// floor mode
 		// life would be easier if the roz we're talking about for complex zoom wasn't setting this as well
 
-		// fprintf(stderr, "Tilemap %d is a floor using :\n", tm);
+		// flogerror(stderr, "Tilemap %d is a floor using :\n", tm);
```
[the remark this answers is not in the document]
Comment (DavidHaywood) [reply src/mame/snk/hng64_v.cpp]: I do, it was a search+replace as you may have gathered ;)

File: src/mame/snk/hng64.cpp (hunk 1 above)
Comment (DavidHaywood) [reply src/mame/snk/hng64.cpp:1725]: I'm not sure, it's just STEP1024(0,4) but with ^4 applied to get the pixels in the right order, I don't know if there's a macro for that.

Comment (happppp) [reply src/mame/snk/hng64.cpp:1725]: I mean something like init_texlayout_xoffset_4() in the driver class constructor.

Comment (DavidHaywood) [reply src/mame/snk/hng64.cpp]: that appears to work, yes

File: src/mame/snk/hng64.cpp [hunk 3]
```diff
@@ -1720,20 +1720,47 @@ static const gfx_layout hng64_16x16x8_spritelayout =
...
+static uint32_t texlayout_xoffset_4[1024];
+static const uint32_t texlayout_yoffset_4[1024] = { STEP1024(0,4096) };
+
+void hng64_state::texlayout_xoffset_4_create()
+{
+	for (int i = 0; i < 1024; i++)
+	{
+		texlayout_xoffset_4[i] = (i * 4) ^ 4;
+	}
+}
```
Comment (cuavas) [inline src/mame/snk/hng64.cpp:1734]: It’s actually possible to generate this at compile time.  I’ll patch it up after merging I guess.

**Pull request merged**